src/ledger/test/LedgerTestUtils.cpp

```diff
@@ -246,6 +257,31 @@ makeValid(ClaimableBalanceEntry& c)
     }
 }
 
+void
+makeValid(LiquidityPoolEntry& lp)
+{
+    auto& cp = lp.body.constantProduct();
+    cp.params.assetA.type(ASSET_TYPE_CREDIT_ALPHANUM4);
+    strToAssetCode(cp.params.assetA.alphaNum4().assetCode, "CAD");
+
+    cp.params.assetB.type(ASSET_TYPE_CREDIT_ALPHANUM4);
+    strToAssetCode(cp.params.assetB.alphaNum4().assetCode, "USD");
+
+    cp.params.fee = 30;
+
+    cp.reserveA = std::abs(cp.reserveA);
```

Comment:
I don't think you need both `abs` and `clampLow`. If `v >= 0` then `clampLow(0, v)` leaves `v` unchanged. This applies to all of the uses here.

Reply:
Updated to just use `abs`.